### ISSUE TITLE
More videos in a single view raises error -11839 "Cannot Decode"

### ISSUE THREAD

**Issue by ugochirico** [casatwy/CTVideoPlayerView#47]
I added several VideoViews in a single view controller.
After about 16 video views (on iPhone 7 with iOS 11.2.5) I got the following error:

Error Domain=AVFoundationErrorDomain Code=-11839 "Cannot Decode" UserInfo={NSLocalizedDescription=Cannot Decode, NSUnderlyingError=0x1c1859fb0 {Error Domain=NSOSStatusErrorDomain Code=-12913 "(null)"}, NSLocalizedRecoverySuggestion=Stop any other actions that decode media and try again., NSLocalizedFailureReason=The decoder required for this media is busy.}

Any suggestions?

**Comment by casatwy** [comment]
does all the 16 video views share 1 url?

**Comment by ugochirico** [comment]
@casatwy no, the urls are all different.
Is there a way to avoid from this error?

**Comment by jagdev7dayz** [comment]
did you get any solution for it?

**Comment by ugochirico** [comment]
No. You cannot load more then 16 videos. The only way is to implement an algo that manages the videos considering such limitation.

**Comment by bstillitano** [comment]
@ugochirico do you have an example of how you solved this?
